app/assets/javascripts/discourse/app/components/sidebar/section.js

```diff
@@ -27,7 +27,22 @@ export default class SidebarSection extends GlimmerComponent {
     }
   }
 
+  @action
+  handleMultipleHeaderActions(id) {
+    this.args.headerActions
+      .find((headerAction) => headerAction.id === id)
+      .action();
+  }
```

Comment:
Hmm can the `headerActions` be an object instead as I think it'll be a better data structure to use here as compared to the array since we need to lookup the header action by id.

Reply:
Do you mean for single action? Array is needed when multiple actions are provided, each button on the dropdown list can have a different action. But of course I can change that for single action we will have object.
```
    {{#if this.isMultipleHeaderActions}}
      <DropdownSelectBox
        @options={{hash icon=@headerActionsIcon placementStrategy="absolute"}}
        @content={{@headerActions}}
        @onChange={{action "handleMultipleHeaderActions"}}
        @class="edit-channels-dropdown"
      />
    {{/if}}
```

Reply:
I have some ideas but let's keep your changes for now because I'm not 100% sure if my idea will work. We can continuee to iterate after merging :)